core/src/main/java/org/mskcc/cbio/portal/scripts/MutationFilter.java

```diff
@@ -109,53 +110,55 @@ public boolean acceptMutation(ExtendedMutation mutation) {
           mutationStatusNoneRejects++;
           return false;
       }
-            
-      // Do not accept Silent or Intronic Mutations
-      if( safeStringTest( mutation.getMutationType(), "Silent" ) ||
-               safeStringTest( mutation.getMutationType(), "Intron" ) ){
-         silentOrIntronRejects++;
-         return false;
-      }
-
+      
       // Do not accept LOH or Wildtype Mutations
       if( safeStringTest( mutation.getMutationStatus(), "LOH" ) ||
                safeStringTest( mutation.getMutationStatus(), "Wildtype" ) ){
          lohOrWildTypeRejects++;
          return false;
       }
-
-		// Do not accept Redacted mutations
-		if (safeStringTest(mutation.getValidationStatus(), "Redacted"))
-		{
-		   redactedRejects++;
-		   return false;
-		}
-
-      // Do not accept 3'UTR or 5' UTR Mutations
-      if( safeStringTest( mutation.getMutationType(), "3'UTR" ) ||
-		  safeStringTest( mutation.getMutationType(), "3'Flank" ) ||
-		  safeStringTest( mutation.getMutationType(), "5'UTR" ) ){
-		  utrRejects++;
-         return false;
+      
+      // Do not accept Redacted mutations
+      if (safeStringTest(mutation.getValidationStatus(), "Redacted")) {
+          redactedRejects++;
+          return false;
       }
       
-      if( safeStringTest( mutation.getMutationType(), "5'Flank" ) ) { 
-            if (whiteListGenesForPromoterMutations.contains(mutation.getEntrezGeneId())){
+      //Filter by types if specified in the meta file, else filter for the default types
+      if (filteredMutations != null) {
+          if (filteredMutations.contains(mutation.getMutationType())) {
+              addRejectedVariant(rejectionMap, mutation.getMutationType());
+              return false;
+          } else {
+              if( safeStringTest( mutation.getMutationType(), "5'Flank" ) ) {
```

Comment:
Jsut for the record: I think that  when a variant_classification_filter *is* specified it should be leading and there should be no whitelist. IMO a whitelist would lead to confusion in this case.